src/generate-query.ts

```diff
@@ -610,6 +638,170 @@ function getArgsAndVars(
   }
 }
 
+/**
+ * Given a field node, return the listSize directive, if it exists
+ */
+function getListSizeDirective(field: FieldDefinitionNode): DirectiveNode {
+  return field?.directives.find((dir) => dir.name.value === 'listSize')
+}
+
+/**
+ * Given a listSize directive, get the slicingArguments argument
+ */
+function getSlicingArguments(listSizeDirective: DirectiveNode): ArgumentNode {
+  if (!listSizeDirective) return undefined
+
+  const slicingArgumentsArg = listSizeDirective.arguments.find(
+    (arg) => arg.name.value === 'slicingArguments'
+  )
+
+  if (!slicingArgumentsArg || slicingArgumentsArg.value.kind !== 'ListValue')
+    return undefined
+
+  return slicingArgumentsArg
+}
+
+/**
+ * Given a listSize directive, get the requireOneSlicingArgument argument
+ */
+function getRequireOneSlicingArgument(
+  listSizeDirective: DirectiveNode
+): ArgumentNode {
+  if (!listSizeDirective) return undefined
+
+  const requireOneSlicingArg = listSizeDirective.arguments.find(
+    (arg) => arg.name.value === 'requireOneSlicingArgument'
+  )
+
+  if (
+    !requireOneSlicingArg ||
+    requireOneSlicingArg.value.kind !== 'BooleanValue'
+  )
+    return undefined
+
+  return requireOneSlicingArg
+}
+
+/**
+ * For a given listSize directive, return a default value for it
+ */
+function getListSizeDirectiveDefaultValue(
+  listSizeDirective: DirectiveNode,
+  typeNode: TypeNode,
+  config: InternalConfiguration,
+  schema: GraphQLSchema
+) {
+  const slicingArgumentsArg = getSlicingArguments(listSizeDirective)
+  if (!slicingArgumentsArg) return undefined
+
+  /**
+   * Already checked slicingArgumentsArguments is a list
+   *
+   * Extract paths from slicingArgumentsArguments
+   */
+  let slicingArgumentsPaths = (slicingArgumentsArg.value as ListValueNode).values.map(
+    (value) => {
+      if (value.kind === 'StringValue') return value.value
+    }
+  )
+
+  const requireOneSlicingArg = getRequireOneSlicingArgument(listSizeDirective)
+  if (
+    !requireOneSlicingArg ||
+    (requireOneSlicingArg.value as BooleanValueNode).value === true
+  ) {
+    slicingArgumentsPaths = slicingArgumentsPaths.slice(0, 1)
+  }
+  const slicingArgumentsTokenizedPaths = slicingArgumentsPaths.map((value) =>
+    value.split('.')
+  )
+
+  if (
+    slicingArgumentsPaths.length === 1 &&
```

Comment:
Why is `slicingArgumentsPaths.length === 1` necessary?

Reply:
If requireOneSlicingArg is false and
```
const slicingArgumentsArg = getSlicingArguments(listSizeDirective)
```
returns no slicing args then the paths could be 0 so we might not be able to index into the first one

Reply:
also if only 1 is in the list and it has only 1 token then we will be returning a simple var value back

Reply:
I was wondering why it isn't `slicingArgumentsPaths.length > 0`. Doesn't this behave strangely if there is more than one path?

Reply:
hmmm... yea I guess there is a disconnect between the slicingargs list (could contain more than one argument root) and the caller which just wants a value for one argument

Reply:
@Alan-Cha, I pushed some new tests that I thought would fail with the code we were discussing (multiple single level slicing arguments, requireOneSlicingArg: false). Surprisingly they passed. 

The multiple single level slicing arguments one works because requireOneSlicingArg is defaulted to true so we only consider a single slicing arg when creating variables.

The requireOneSlicingArg: false works because an earlier call to getMissingSlicingArgs does not return them since setting it to false means they are never required.

So I think there is some dead code in there, but it all works based on what I have thought of. Let me know if you think there is a test that will break it.

Reply:
@Alan-Cha I addressed this specific comment in the latest push. I will mark this conversation as resolved.